src/utils.ts

```diff
@@ -10,6 +12,10 @@ export function debounce(fn, wait = 0) {
     };
 }
 
+export const routerPush = (router: Router, path: string) => {
+    debounce(() => router.push(path), 75)();
```

Comment:
75 ms seems to work well for showing loading spinner, having no timeout wasn't working well with admin level toggle so thought to extract this out so that it is consistent with all the `router.push`'s